examples/indexing/fancy_indexing.py

```diff
@@ -0,0 +1,67 @@
+import numpy as np
+from numpy.random import randint
+
+# %out = tensor.gather %input[%indices] gather_along_dimension(1) :
+#   tensor<2x3x4xf32>[tensor<5x6x 1xindex>] -> tensor<5x6x2x1x4xf32>
+
+input = randint(0, 10, (2, 3, 4))
+indices = randint(0, 2, (5, 6))
+out = input[:, indices, :]
+out_ = np.moveaxis(out, (1, 2), (0, 1))
```

Comment:
How about asserting the result shape here? Mainly for documentation purpose such that one can see it without executing the code...